packages/common/src/type-helpers.ts

```diff
@@ -141,8 +141,35 @@ export function SymbolBasedInstanceOfError<E extends Error>(markerName: string):
   };
 }
 
-// Thanks MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze
-export function deepFreeze<T>(object: T): T {
+// This implementation is based on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze.
+// Note that there are limits to this approach, as traversing using getOwnPropertyXxx doesn't allow
+// reaching variables defined using internal scopes. This implementation specifically look for and deal
```

Comment:
```suggestion
// reaching variables defined using internal scopes. This implementation specifically looks for and deals
```

Reply:
Why is dealing with these types important? Have you seen any shared maps and sets that are exposed to users?